CRM/Case/Form/Activity.php

```diff
@@ -657,4 +691,68 @@ public function postProcess($params = NULL) {
     }
   }
 
+  /**
+   * Returns the groups that contacts must belong to in order to be assigned
+   * an activity for this case. It returns an empty array if no groups are found for
+   * the case type linked to the caseId.
+   *
+   * @return array
+   */
+  private function getActivityAssignmentGroups() {
+    if (!$this->_caseTypeDefinition) {
+      return [];
+    }
+
+    $assignmentGroups = [];
+    foreach ($this->_caseTypeDefinition as $caseId => $definition) {
+      if (!empty($definition['activityAsgmtGrps']) && $definition['restrictActivityAsgmt'] == self::RESTRICT_ASSIGNMENT_BY_GROUP) {
+        $assignmentGroups = array_merge($assignmentGroups, $definition['activityAsgmtGrps']);
+      }
+    }
+
+    return $assignmentGroups;
+  }
+
+  /**
+   * Returns whether contacts must have a user account in order to be
+   * assigned an activity for this case.
+   *
+   * @return bool
+   */
+  private function restrictAssignmentByUserAccount() {
+    if (!$this->_caseTypeDefinition) {
+      return FALSE;
+    }
+
+    foreach ($this->_caseTypeDefinition as $caseId => $definition) {
+      if (!empty($definition['restrictActivityAsgmt']) && $definition['restrictActivityAsgmt'] == self::RESTRICT_ASSIGNMENT_BY_USER_ACCOUNT) {
+        return TRUE;
+      }
+    }
+
+    return FALSE;
+  }
```

Comment:
I would suggest making this function more generic - rename it to `restrictAssignment` and have it return the value of `restrictActivityAsgmt` (0, 1, or 2)